recommender/static/js/src/recommender.js

```diff
@@ -1,852 +1,930 @@
+                        }
+                    }
+                    paginationItem();
+                    pagination();
+                    backToView();
+                    Logger.log('mit.recommender.importResource', generateLog(loggerStatus['importResource']['complete'], data));
+                },
+            });
+        });
+    }
+
+    /**
+     * Switch from pages for resource addition, edit, flag, etc. to pages for viewing resources.
+     */
+    function backToView() {
+        modals = [
+            '.recommenderModify',
+            '.flagResourcePage',
+            '.editResourcePage',
+            '.addResourcePage',
+            '.deendorsePage',
+            '.endorsePage',
+            '.importResourcePage'
+        ]
+        for (i = 0; i < modals.length; i++){
+            $(modals[i], element).hide().attr('aria-hidden', 'true');
+        }
+
+        if ($('.recommenderResource', element).length == 0) {
+            $('.noResourceIntro', element).show().attr('aria-hidden', 'false');
+        }
+        $('.recommenderResource', element).removeClass('resourceHovered');
+        $('.previewingImg', element).hide().attr('aria-hidden', 'true');
+        $('.descriptionText', element).hide().attr('aria-hidden', 'true');
         if (!DISABLE_DEV_UX) {
-            $('.show_problematic_reasons', element).addClass('hidden');
-            $('.show_endorsed_reasons', element).addClass('hidden');
+            $('.showProblematicReasons', element).hide().attr('aria-hidden', 'true');
+            $('.showEndorsedReasons', element).hide().attr('aria-hidden', 'true');
         }
-		$('.recommender_content', element).show();
-	}
-
-    /* Trigger event of mode switching from resource addition/edit/flag/staff-edit to resource list displaying. */
-	$('.backToViewButton', element).click(function() {
-        var divs = $('.flagResourceBlock, .editResourceBlock, .addResourceBlock, .deendorseBlock, .endorseBlock', element);
-        function findDisplayedBlock() {
-            for (var key in divs) {
-                if ($(divs[key]).attr('style') != "display: none;") { return divs[key]; }
+        $('.recommenderContent', element).show().attr('aria-hidden', 'false');
+        $('.hideShow').focus();
+    }
+
+    /**
+     * Log the typed information for an incomplete submission.
+     * @param {string} selector The string indicating the element we are going to select and log its value.
+     * @param {element} activePage The element which is currently shown to learners.
+     * @param {dictionary} logStudentInput The information which will be logged.
+     * @returns {dictionary} The information which will be logged.
+     */
+    function logTypedInformation(selector, activePage, logStudentInput) {
+        if ($(selector, activePage).length != 0) {
+            $(selector, activePage).each(function() {
```

Comment:
You don't even need to check the length in the line above. Just iterate over the jQuery object. If the length is zero, then the function will never be called, which is what you want.